app/models/aggregated_cash_income.rb

```diff
@@ -0,0 +1,106 @@
+class AggregatedCashIncome
+  include ActiveModel::Model
+
+  CATEGORIES= %i[
+      benefits
+      friends_or_family
+      maintenance_in
+      property_or_lodger
+      pension
+  ].freeze
+
+  RANGE = (1..3).freeze
+
+  def self.category_attributes(categories)
+    categories.each do |category|
+      attr_accessor "check_box_#{category}".to_sym
+      RANGE.each do |i|
+        attr_accessor "#{category}_month_#{i}".to_sym
+      end
+    end
+  end
+
+  attr_accessor :legal_aid_application_id,
+                :category,
+                :none_selected
+
+  category_attributes CATEGORIES
+
+  validate :check_category
+  validate :anything_selected?
+
+  def checkbox_integrity(params)
+    CATEGORIES.each do |category|
+      checkbox_selected = params["check_box_#{category}"] == 'true'
+
+      if checkbox_selected
+        @category = category
+        check_category
+      else
+        args = search_args(category)
+        CashTransaction.where(**args).destroy_all
+      end
+    end
+
+    anything_selected?
+  end
+
+  def save(params)
+    params.each do |key, value|
+      __send__("#{key}=",value) if value != ''
+    end
+
+    checkbox_integrity(params)
+
+    return false unless errors.empty?
+
+    save_cash_transaction_records
+  end
+
+  private
+
+  def save_cash_transaction_records
+    CATEGORIES.each do |category|
+      save_category(category) if checkbox_for(category) == 'true'
+    end
+  end
+
+  def save_category(category)
+    RANGE.each do |i|
+      args = search_args(category)
+      amount = __send__("#{category}_month_#{i}".to_sym)
+      date = calculated_date(i)
+
+      CashTransaction.where(**args, transaction_date: date).destroy_all
+      CashTransaction.create!(**args, transaction_date: date, amount: amount)
+    end
```

Comment:
I don't think I like the use of **args here - it over complicates things.  And I think I'd rather see existing records being updated rather than deleted and re-created.  so how about something like this:
```
CashTransaction.update_or_create(legal_aid_application_id: legal_aid_application_id,
                                                             transaction_type_id: TransactionType.credits.find_by(name: category).id,
                                                             date: date,
                                                             amount: amount)
```

Then writing a method  the `.update_or_create` method on CashTransaction to find the record and update it or create it if it doesn't exist.

I think the intent is a lot clearer.

Also, don't forget that we have to delete all records belonging to categories that are not entered if they exist.

Reply:
So we had a discussion about updating records, we thought if a user saves and returns that then, not only amount, but date would have to be updated as well. Due to that we thought it easier to wipe the records and recreate them.

`Also, don't forget that we have to delete all records belonging to categories that are not entered if they exist.`

Yes we are doing that on line 46.